opengovsg/plumber
# PLU-80: [SNAZZY-VARIABLES-TEMP-FIX] Show variable values in PowerInput

## Problem
In addition to labels, we also want to show the variable's value when our user edits a pipe:

![Screenshot 2023-08-14 at 10 08 06 AM](https://github.com/opengovsg/plumber/assets/135598754/bc73c8fe-1649-45e4-9b87-1705e50ee2bb)

### The long term solution is to migrate away from `PowerInput`.
But that requires some rework + integrating tip-tap, which will take a while. We need a temporary solution in the meantime.

## Solution
We will enhance `PowerInput` to display values inside the variable pills. The high level idea is to enhance `VariableLabelMap` to also store test run values, and extract those values when rendering variable pills.

There is also one interesting thing I found while fixing types inside `PowerInput`:
* `VariableList` (fka `Suggestions`) does not use Slate's `VariableElement` at all; it directly reads from `priorStepsWithExecutions` instead.
* This is why we can convert `Pick<VariableElement, 'name' | 'value'>` to `Variable` inside `handleVariableSuggestionClick` without things exploding.

**BEFORE**
<img width=600 src="https://github.com/opengovsg/plumber/assets/135598754/5f2cb9a3-e276-4cf9-9fd7-89cdcc6b4b9d" />

**AFTER**
<img width=600 src="https://github.com/opengovsg/plumber/assets/135598754/64df40b7-30e8-4923-bca3-eb9de66999b3" />

## Tests
- Check that value shows up inside the variable badge.
- Check that if value is falsy (e.g. unfilled optional form field), value is not shown.
- Regression test: check that editing old pipes still work.
